language-bindings/python/src/wamr/wamrapi/wamr.py

```diff
@@ -129,21 +179,53 @@ def _create_module_inst(self, module: Module, stack_size: int, heap_size: int) -
         return module_inst
 
 
+id_to_exec_env_mapping = {}
+
+
 class ExecEnv:
-    def __init__(self, module_inst: Instance, stack_size: int = 65536):
+    def __init__(self, module_inst: Instance | None = None, stack_size: int = 65536):
         self.module_inst = module_inst
         self.exec_env = self._create_exec_env(module_inst, stack_size)
+        self.own_c = True
+
+        id_to_exec_env_mapping[str(addressof(self.exec_env.contents))] = self
 
     def __del__(self):
-        print("deleting ExecEnv")
-        wasm_runtime_destroy_exec_env(self.exec_env)
+        if self.own_c:
+            print("deleting ExecEnv")
+            wasm_runtime_destroy_exec_env(self.exec_env)
```

Comment:
Should we also remove related element from id_to_exec_env_mapping?

Reply:
done here: https://github.com/bytecodealliance/wasm-micro-runtime/pull/2449/commits/e224b300a9843319abae63ea1de1c489161f74ee